pkg/sdk/sdk_test.go

```diff
@@ -0,0 +1,518 @@
+						t.Error(err.Error())
+						return
+					}
+					if !reflect.DeepEqual(id, "1") {
+						t.Errorf("Want: %v, Got: %v", "1", id)
+					}
+					if !reflect.DeepEqual(fileBytes, []byte("abc")) {
+						t.Errorf("Want: %v, Got: %v", []byte("abc"), fileBytes)
+					}
+					defer file.Close()
+					cType := r.Header.Get("Content-Type")
+					x := strings.Split(cType, ";")
+					if !reflect.DeepEqual(x[0], "multipart/form-data") {
+						t.Errorf("Want: %v, Got: %v", "multipart/form-data", x[0])
+					}
+					response.ToJson(w, http.StatusOK, "SUCCESS", map[string]interface{}{
+						"id": id,
+					})
+				})
+				return svr
+			},
+			ValidateFunc: func(err error) {
+				if err != nil {
+					t.Errorf("Want: %v, Got: %v", nil, err)
+				}
+			},
+			cleanupFunc: func(svr *httptest.Server) {
+				svr.Close()
+			},
+		},
+		{
+			name: "Failure:: Replace :: type inference fail",
+			id:   "1",
+			setupFunc: func() *httptest.Server {
+				svr := testServer("/v1/register/{id}", http.MethodPut, func(w http.ResponseWriter, r *http.Request) {
+					response.ToJson(w, http.StatusOK, "SUCCESS", Reader(""))
+				})
+				return svr
+			},
+			ValidateFunc: func(err error) {
+				if err.Error() != "unable to parse response data" {
+					t.Errorf("Want: %v, Got: %v", "unable to parse response data", err)
+				}
+			},
+			cleanupFunc: func(svr *httptest.Server) {
+				svr.Close()
+			},
+		},
+		{
+			name: "Failure:: Replace :: json failure",
+			id:   "1",
+			setupFunc: func() *httptest.Server {
+				svr := testServer("/v1/register/{id}", http.MethodPut, func(w http.ResponseWriter, r *http.Request) {
+
+					json.NewEncoder(w).Encode("")
+				})
+				return svr
+			},
+			ValidateFunc: func(err error) {
+				if err.Error() != "json: cannot unmarshal string into Go value of type model.Response" {
+					t.Errorf("Want: %v, Got: %v", "json: cannot unmarshal string into Go value of type model.Response", err)
+				}
+			},
+			cleanupFunc: func(svr *httptest.Server) {
+				svr.Close()
+			},
+		},
+		{
+			name: "Failure:: Replace :: incorrect id in response",
+			id:   "1",
+			setupFunc: func() *httptest.Server {
+				svr := testServer("/v1/register/{id}", http.MethodPut, func(w http.ResponseWriter, r *http.Request) {
+					response.ToJson(w, http.StatusOK, "SUCCESS", map[string]interface{}{
+						"id": "gfhgv",
+					})
+				})
+				return svr
+			},
+			ValidateFunc: func(err error) {
+				if err.Error() != "incorrect id received in response" {
+					t.Errorf("Want: %v, Got: %v", "incorrect id received in response", err)
+				}
+			},
+			cleanupFunc: func(svr *httptest.Server) {
+				svr.Close()
+			},
+		},
+		{
+			name: "Failure:: Replace :: incorrect status code",
+			setupFunc: func() *httptest.Server {
+				svr := testServer("", http.MethodPut, func(w http.ResponseWriter, r *http.Request) {})
+				return svr
+			},
+			ValidateFunc: func(err error) {
+				if err.Error() != "non success status code received : 404" {
```

Comment:
Where is the non-success status code(404) sent from the server?
Assuming this test cases tests the scenario when the server responds with non 200 status code.
Explicitly send the status code from the test server and compare the same code in the `ValidateFunc`

Reply:
in this case we are not sending any response from server whereas in this case the path to server is incorrect so it automatically state that server not found with 404